IdGenTests/IdGeneratorTests.cs

```diff
@@ -78,16 +78,34 @@ public void GeneratorId_ShouldBeMasked_WhenReadFromProperty()
         Assert.AreEqual((1 << g.Options.IdStructure.GeneratorIdBits) - 1, g.Id);
     }
 
+    [TestMethod]
+    public void Constructor_DoesNotThrow_OnMaxGeneratorId()
+    {
+        var structure = new IdStructure(41, 10, 12);
+        int maxgeneratorid = (int)GetMaxGeneratorId(structure);
+        new IdGenerator(maxgeneratorid, new IdGeneratorOptions(structure));
+    }
+
+    [TestMethod]
+    public void Constructor_DoesNotThrow_OnGeneratorId_0()
+    {
+        new IdGenerator(0, new IdGeneratorOptions(new IdStructure(41, 10, 12)));
+    }
+
     [TestMethod]
     [ExpectedException(typeof(ArgumentNullException))]
     public void Constructor_Throws_OnNull_Options()
         => new IdGenerator(1024, null!);
 
-
     [TestMethod]
     [ExpectedException(typeof(ArgumentOutOfRangeException))]
-    public void Constructor_Throws_OnInvalidGeneratorId_Positive()
-        => new IdGenerator(1024, new IdGeneratorOptions(new IdStructure(41, 10, 12)));
+    public void Constructor_Throws_OnInvalidGeneratorId_Positive_MaxPlusOne()
+    {
+        var structure = new IdStructure(41, 10, 12);
+        var maxgeneratorid = GetMaxGeneratorId(structure);
```

Comment:
Just a little nitpick: I'd hardcode this to the actual value; if there's a bug in the ("copied" or "lifted") `GetMaxGeneratorId` code then the bug will also be in this test. So I guess I'm saying I'd set `maxgeneratorid` to `1024` here and maybe even add a little comment making clear it's related to the `10` in the `IdStructure`.

But, again, thanks for the PR. I can make this (tiny) change myself later when I get back if you want to.

Reply:
That's fair. I updated the PR for that